benchmarks/doneahuerta/doneahuerta.cc

```diff
@@ -0,0 +1,375 @@
+        }
+        /**
+         * @}
+         */
+
+        /**
+         * @name Qualitative properties one can ask a material model
+         * @{
+         */
+
+        /**
+         * Return whether the model is compressible or not.
+         * Incompressibility does not necessarily imply that the density is
+         * constant; rather, it may still depend on temperature or pressure.
+         * In the current context, compressibility means whether we should
+         * solve the continuity equation as $\nabla \cdot (\rho \mathbf u)=0$
+         * (compressible Stokes) or as $\nabla \cdot \mathbf{u}=0$
+         * (incompressible Stokes).
+         */
+        virtual bool is_compressible () const;
+        /**
+         * @}
+         */
+        /**
+         * Declare the parameters this class takes through input files.
+         */
+        static
+        void
+        declare_parameters (ParameterHandler &prm);
+
+        /**
+         * Read the parameters this class declares from the parameter file.
+         */
+        virtual
+        void
+        parse_parameters (ParameterHandler &prm);
+
+
+
+        /**
+         * @name Reference quantities
+         * @{
+         */
+        virtual double reference_viscosity () const;
+        /**
+         * @}
+         */
+    };
+
+
+    template <int dim>
+    double
+    DoneaHuertaMaterial<dim>::
+    reference_viscosity () const
+    {
+      return 1.;
+    }
+
+
+    template <int dim>
+    bool
+    DoneaHuertaMaterial<dim>::
+    is_compressible () const
+    {
+      return false;
+    }
+
+    template <int dim>
+    void
+    DoneaHuertaMaterial<dim>::declare_parameters (ParameterHandler &)
+    {
+    }
+
+
+    template <int dim>
+    void
+    DoneaHuertaMaterial<dim>::parse_parameters (ParameterHandler &)
+    {
+      // Declare dependencies on solution variables
+      this->model_dependence.viscosity = MaterialModel::NonlinearDependence::none;
+      this->model_dependence.density = MaterialModel::NonlinearDependence::none;
+      this->model_dependence.compressibility = MaterialModel::NonlinearDependence::none;
+      this->model_dependence.specific_heat = MaterialModel::NonlinearDependence::none;
+      this->model_dependence.thermal_conductivity = MaterialModel::NonlinearDependence::none;
+    }
+
+
+    template <int dim>
+    DoneaHuertaBoundary<dim>::DoneaHuertaBoundary ()
+    {}
+
+
+    template <>
+    Tensor<1,2>
+    DoneaHuertaBoundary<2>::
+    boundary_velocity (const types::boundary_id ,
+                       const Point<2> &p) const
+    {
+      return AnalyticSolutions::DoneaHuerta_velocity (p);
+    }
+
+
+    template <>
+    Tensor<1,3>
+    DoneaHuertaBoundary<3>::
+    boundary_velocity (const types::boundary_id ,
+                       const Point<3> &) const
+    {
+      Assert (false, ExcNotImplemented());
+      return Tensor<1,3>();
+    }
+
+
+
+
+
+
+    /**
+     *gravity model for the DoneaHuerta benchmark
+    */
```

Comment:
Add a space before 'gravity', capitalize it, and make indentation consistent

Reply:
done